src/scales/scale.linear.js

```diff
@@ -11,123 +11,129 @@ var defaultConfig = {
 	}
 };
 
+var DEFAULT_MIN = 0;
+var DEFAULT_MAX = 1;
+
+function getMatchingVisibleMetas(scale) {
+	var isHorizontal = scale.isHorizontal();
+
+	return scale.chart._getSortedVisibleDatasetMetas()
+		.filter(function(meta) {
+			return isHorizontal ? meta.xAxisID === scale.id : meta.yAxisID === scale.id;
+		});
+}
+
+function getStack(stacks, stacked, meta) {
```

Comment:
It looks like this method might only be called once for each stack, so I'm wondering if we need this caching behavior. If we don't need to cache then I'd probably call it `createStack`. If we do need the caching behavior then it might be clearer to call it `getOrCreateStack`

Reply:
Function renamed (its called once per stack per dataset)

Reply:
If it's only called once should we remove the caching mechanism? I think that would make the code much simpler

Reply:
Its called once in each dataset for each stack and the stack needs to be same. So it needs to be cached. Master has same logic, here its just extracted to function.